packages/@lwc/engine/src/framework/api.ts

```diff
@@ -142,9 +146,12 @@ const ElementHook: Hooks<VElement> = {
 const CustomElementHook: Hooks<VCustomElement> = {
     create: (vnode) => {
         const { sel } = vnode;
-        vnode.elm = document.createElement(sel);
-        linkNodeToShadow(vnode);
-        createViewModelHook(vnode);
+        const elm = document.createElement(sel);
+
+        linkNodeToShadow(elm, vnode);
+        createViewModelHook(elm, vnode);
+        vnode.elm = elm;
```

Comment:
Until now, the element was attached to the `VNode` regardless if the custom element creation was successful or not. Because of this, it would require `resetShadowRoot` the `resetShadow` root method to check if the element is connected or not to the parent before attempting to remove it. Which is slow.

By attaching the created element after the custom element is created, we can ensure that the element is connected to its parent.